src/server/db_slice.h

```diff
@@ -307,6 +307,8 @@ class DbSlice {
                                        unsigned req_obj_type) const;
 
   OpResult<ItAndUpdater> AddOrFind(const Context& cntx, std::string_view key);
+  OpResult<ItAndUpdater> AddOrFind(const Context& cntx, std::string_view key,
```

Comment:
I do not know if it justified another overload here. you could check the return type in the only place you use `AddOrFind` like this.